sdk/dapp-kit/src/components/styling/InjectedThemeStyle.tsx

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { assignInlineVars } from '@vanilla-extract/dynamic';
+
+import { styleDataAttributeSelector } from '../../constants/styleDataAttribute.js';
+import { themeVars } from '../../themes/themeContract.js';
+import type { Theme, ThemeVars } from '../../themes/themeContract.js';
+
+type InjectedThemeStyleProps = {
+	theme: Theme;
+};
+
+export function InjectedThemeStyle({ theme }: InjectedThemeStyleProps) {
+	const defaultStyles = `${styleDataAttributeSelector}{${cssStringFromTheme(
+		'light' in theme ? theme.light : theme,
+	)}}`;
+	const darkModeStyles =
```

Comment:
We probably want to let folks customize when the dark mode theme applies. A lot of folks use class-based dark mode.

Reply:
We also probably want to support defining the theme directly too.